google/cloud/security/enforcer/gce_firewall_enforcer.py

```diff
@@ -486,6 +486,9 @@ def add_rules(self, rules, network_name=None):
           InvalidFirewallRuleError: One or more rules failed validation.
         """
         for rule in rules:
```

Comment:
Ahh, I missed that this was to check for an invalid policy.

Okay, so two things. One, add_rule is what cares that rule is a dict, so the isinstance check belongs there, in case some other code path is calling add_rule directly. The add_rules methods cares that rules is an iterable of rule dicts, so slightly different flavor, and different check.

It also makes sense to verify that the loaded json policy in enforcer.py is in the expected format and return a meaningful error to the user at that point, rather than waiting for it to error out while building the rules object.